### PR TITLE
net: ipv4: Make native IPv4 options dependent on NET_NATIVE_IPV4

### PR DESCRIPTION
**Pull request by rlubos**
IPv4 Kconfig options which only affect native IPv4 stack should be dependent on `NET_NATIVE_IPV4`, similarly as it's done for IPv6.

### CODE REVIEW

**Comment by rlubos** [comment]
@jukkar Currently this patch follows what's been done for IPv6:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/Kconfig.ipv6

I've noticed one inconsequence with this approach though, hence marking this as a draft for now. 

According to Kconfig, the IPv4/6 address counts does not depend on native IP networking. Following this logic, it should be possible to register IP addresses even for offloaded interfaces, which IMO makes sense, as for example we should be able then to display them with net shell.

However, when I look how those IPv4/6 address containers are defined, they only seem to be enabled if native networking is used:
https://github.com/zephyrproject-rtos/zephyr/blob/119b238594971a41dc55609d7509677efc675a6c/subsys/net/ip/net_if.c#L98

Shouldn't those rather depend on `CONFIG_NET_IPV4/6` instead, so that it's still possible to register IP addresses in purely offloaded builds? I'm pretty sure registering IP addresses for offloaded interfaces works, as we do that for nRF91 in some NCS samples. It requires though native networking support enabled, which IMO shouldn't be the case.

I can investigate this further if we agree this should be changed.

**Comment by jukkar** [comment]
Good point, I agree, we can have the addresses for offloaded interfaces

**Review by jukkar — DISMISSED** [review]

**Comment by rlubos** [comment]
Ok, I have updated `net_if.c` ifdefs so that it's now possible to register IP addresses even if native IP stack is disabled. This allows for example to use connection manager features w/o a need to enable native stack in purely offloaded scenarios. I think it's ready to review.

I've moved a few functions in the file (no functional changes just relocating the code) to avoid further ifdef-slicling of the file. Alternatively, I can keep the code as is, and introduce more ifdeffing in case that's preferred.

**Review by pdgendt — DISMISSED** [review]

**Comment by rlubos** [comment]
Latest push with CI fixes.

**Comment by jukkar** [comment]
There are also checks for native IPv4/6 in `ipv4.c` and `ipv6.c` in net-shell, perhaps those should be adjusted too.

File: subsys/net/lib/shell/iface.c [hunk 1]
```diff
@@ -148,17 +147,17 @@ static const char *iface_flags2str(struct net_if *iface)
...
 static void iface_cb(struct net_if *iface, void *user_data)
 {
-#if defined(CONFIG_NET_NATIVE)
 	struct net_shell_user_data *data = user_data;
 	const struct shell *sh = data->sh;
 
-#if defined(CONFIG_NET_IPV6)
-	struct net_if_ipv6_prefix *prefix;
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if_ ipv6_prefix *prefix;
```
Comment (pdgendt) [inline subsys/net/lib/shell/iface.c:157]: Looks like a typo

**Comment by rlubos** [comment]
> There are also checks for native IPv4/6 in `ipv4.c` and `ipv6.c` in net-shell, perhaps those should be adjusted too.

Good point @jukkar, I've modified `net iface/ipv4/ipv6` commands so that they can work and print statuses with native IP stack disabled. Although, to make net shell functional w/o native stack, we need your shell PR merged first, so that the disabled shell modules are not compiled in, currently some of them throw errors if built w/o native stack.

File: subsys/net/lib/shell/iface.c (hunk 1 above)
Comment (rlubos) [reply subsys/net/lib/shell/iface.c]: yep, thanks!

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**